File: backend/core/utils/helper/FilterHelper.py
from bson import ObjectId
from datetime import datetime
from typing import List

class FilterHelper:

    @staticmethod
    def idFilter(imageId: str):
        return {
            '_id': ObjectId(imageId)
        }

    @staticmethod
    def pathFilter(path: str):
        return {
            'path': path
        }

    @staticmethod
    def tagFilter(tag: List[str] | str):
        if (isinstance(tag, str)):
            tag = [tag]

        return {
            'tags': {
                '$in': tag
            }
        }

    @staticmethod
    def acquisitionTimeFilter(acquisitionTime: datetime | None):
        if (acquisitionTime is None):
            return {}

        return {
            'timeNode.acquisitionTime': {
                '$gte': acquisitionTime
            }
        }
    
    @staticmethod
    def satelliteFilter(satellite: str | None):
        if (satellite is None):
            return {}

        return {
            'satellite': satellite
        }
    
    @staticmethod
    def sensorFilter(sensor: str | None):
        if (sensor is None):
            return {}

        return {
            'sensor': sensor
        }

File: backend/src/core/utils/helper/FilterHelper.ts
import { ObjectId } from "bson";
import { Filter } from "mongodb";
import { Image } from "../type/data/Image";

export class FilterHelper {

    public static idFilter(id: string): Filter<Image> {
        return { _id: new ObjectId(id) }
    }

    public static pathFilter(path: string): Filter<Image> {
        return { path: path }
    }

    public static tagFilter(tag: string | string[]): Filter<Image> {
        if (Array.isArray(tag)) {
            return { tags: { $all: tag } }
        }

        return { tags: tag }
    }

    public static acquisitionTimeFilter(acquisitionTime: Date): Filter<Image> {
        return {
            'timeNode.acquisitionTime': {
                '$gte': acquisitionTime
            }
        }
    }

    public static satelliteFilter(satellite: string): Filter<Image> {
        return { 'stats.satellite': satellite }
    }

    public static sensorFilter(sensor: string): Filter<Image> {
        return { 'stats.sensor': sensor }
    }
}